url_builder_integration.py

```diff
@@ -0,0 +1,443 @@
+    '''
+    journaldictionary={'arxiv':arxiv_url(search_word,limit),
+                       'preprint':preprint_url(search_word,limit),
+                       'biorxiv':biorxiv_url(search_word,limit),
+                       'medrxiv':medrxiv_url(search_word,limit),
+                       'pbmc':pbmc_url(search_word,limit),
+                       'scielo':scielo_url(search_word,limit)
+                       }
+    return journaldictionary[journal_name]
+
+  
+
+
+  def _url_collector(self):
+    '''
+      Method to scrape the article URLs
+    '''
+    if self.page_num is not None:
+      self.page_url=self._url_base[self.journal]+str(self.page_num)
+    else:
+      self.page_url=self._url_base[self.journal]
+    articles = mining.Element("css_selector",self._anchor_element[self.journal]).get_attribute('href')
+    self.urls = mining.CollectURLs(self.page_url, articles, driver_path=self._driver_path)  
+  
+
+  
+  
+  def _create_url_schema(self,list_of_urls,catalog_url,is_pdf=None,lang='en'):
+    '''
+      Method to create a dataframe.
+
+      Args:
+        list_of_urls(list): URLs scraped.
+        url(str): catalog URL
+        is_pdf: Function to determine if the URL is PDF or not.
+        lang(str): language of the data.
+
+      Returns:
+        Dataframe
+    '''
+    self._url_schema=pd.DataFrame({'article_url': list_of_urls})
+    self._url_schema['catalog_url']=catalog_url
+    if is_pdf==None:
+      self._url_schema['is_pdf'] =0
+    else:
+      self._url_schema['is_pdf'] = self._url_schema['article_url'].apply(is_pdf)
+    self._url_schema['language']=lang
+    self._url_schema['status']='not mined'
+    self._url_schema['timestamp']= datetime.utcnow()
+    self._url_schema['worker_id']=None
+    self._url_schema['meta_info']=None
+    return self._url_schema
+  
+  
+ 
+
+    
+  def _send_to_bigquery(self):
+    '''
+        Method to push urls to bigquery.
+    '''
+    credentials = service_account.Credentials.from_service_account_file(self.credentials)
+    pandas_gbq.to_gbq(self._url_schema, self.table_id, project_id=self.project_id, if_exists='append', table_schema=self._schema,credentials=credentials)
+    pass
+  
+
+
+  def get_urls(self,project_id,table_id):
+    '''
+      To be implemented by the corresponding subclasses
+    '''
+    raise NotImplementedError("Subclasses should implement this!")
+  
+  
+
+
+
+
+
+class arxiv_url(URL_builder):
+  """
+      Child class for getting URLs from arxiv journal.
+      Arxiv: http://export.arxiv.org/
+
+      Args:
+        search_word(str): word to be searched
+        limit(int): number of URLs to be collected
+  """
+
+  def __init__(self,search_word,limit):
+        super().__init__(search_word)  
+        self.limit=limit
+  
+
+  def get_urls(self):
+    '''
+      Method to get URLs and send url_schema to bigquery.
+
+      Returns: url_schema
+
+    '''
+    for self.page_num in range(0,25*(99999),25):
```

Comment:
consider moving `get_urls` method to the base class, because all classes except `scielo_url` class uses the same code/logic in `get_urls` method with only difference in `range` count.